src/storage/src/hummock/iceberg_compactor_runner.rs

```diff
@@ -0,0 +1,118 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use ic_core::CompactionConfig;
+use ic_core::compaction::{Compaction, CompactionType};
+use iceberg::{Catalog, TableIdent};
+use risingwave_connector::sink::iceberg::IcebergConfig;
+use risingwave_pb::iceberg_compaction::IcebergCompactionTask;
+use thiserror_ext::AsReport;
+use tokio::sync::oneshot::Receiver;
+
+use super::HummockResult;
+use crate::hummock::HummockError;
+
+const MAX_SIZE_PER_PARTITION: u32 = 1024 * 1024 * 1024; // 1GB
+
+pub struct IcebergCompactorRunner {
+    pub task_id: u64,
+    pub catalog: Arc<dyn Catalog>,
+    pub table_ident: TableIdent,
+}
+
+impl IcebergCompactorRunner {
+    pub async fn new(iceberg_compaction_task: IcebergCompactionTask) -> HummockResult<Self> {
+        let IcebergCompactionTask { task_id, props } = iceberg_compaction_task;
+        let config = IcebergConfig::from_btreemap(BTreeMap::from_iter(props.into_iter()))
+            .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
+        let catalog = config
+            .create_catalog()
+            .await
+            .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
+        let table_ident = config
+            .full_table_name()
+            .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
+        Ok(Self {
+            task_id,
+            catalog,
+            table_ident,
+        })
+    }
+
+    pub async fn calculate_task_parallelism(&self) -> HummockResult<u32> {
+        let table = self
+            .catalog
+            .load_table(&self.table_ident)
+            .await
+            .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
+        let manifest_list = table
+            .metadata()
+            .current_snapshot()
+            .unwrap()
+            .load_manifest_list(table.file_io(), table.metadata())
+            .await
+            .unwrap();
+
+        let mut all_data_file_size: u32 = 0;
+        for manifest_file in manifest_list.entries() {
+            let a = manifest_file.load_manifest(table.file_io()).await.unwrap();
+            let (entry, _) = a.into_parts();
+            for i in entry {
+                match i.content_type() {
+                    iceberg::spec::DataContentType::Data => {
+                        all_data_file_size += i.data_file().file_size_in_bytes() as u32;
+                    }
+                    iceberg::spec::DataContentType::EqualityDeletes => {}
+                    iceberg::spec::DataContentType::PositionDeletes => {}
+                }
+            }
+        }
+
+        Ok((all_data_file_size / MAX_SIZE_PER_PARTITION) + 1)
+    }
+
+    pub async fn compact_iceberg(self, shutdown_rx: Receiver<()>, parallelism: usize) {
+        let compact = async move {
+            let compaction_config = Arc::new(CompactionConfig {
```

Comment:
We must consider `target_file_size`, or add a doc and implement it in the next PR.